src/cirrus/lib2/eventdb.py

```diff
@@ -51,37 +44,25 @@ def __init__(
     def enabled(self) -> bool:
         return bool(self.event_db_name and self.event_table_name)
 
+    @classmethod
+    def _payload_id_to_record_data(cls, payload_id: str) -> Tuple[str, str, str]:
+        pattern = r"(?P<collections>.+)/workflow-(?P<workflow>[^/]+)/(?P<itemids>.+)"
+        expr = re.compile(pattern)
```

Comment:
Seems like you could move this out to a constant so it doesn't have to be recompiled on every function invocation.